SignalServiceKit/src/Util/JobQueue.swift

```diff
@@ -109,7 +109,11 @@ public extension JobQueue {
         jobRecord.save(with: transaction)
 
         transaction.addCompletionQueue(.global()) {
-            self.workStep()
+            AppReadiness.runNowOrWhenAppDidBecomeReady {
```

Comment:
Being a bit more defensive here.

Reply:
But this is `JobQueue`, not `MessageSenderJobQueue`.  I could see us queuing jobs before the app is ready, e.g. I could see a "migration" enqueuing a "durable account attributes update", for example.

I'm adding an assert `MessageSenderJobQueue`, but I could even see a migration enqueuing a sync message? Maybe?